docs/ldml/tr35-keyboards.md

```diff
@@ -1348,206 +1343,503 @@ This example shows some of the mappings for a French keyboard layout:
+    These refer to groups captured as a set, and can be referenced with the `$1` through `$9` operators in the `to=` pattern. May not be nested.
 
-When the transform is applied, the string matched by the `@from` attribute is replaced by the string in the `@to` attribute. After the change, the current position is reset to just after the text output from the `@to` attribute.
+- **Nested non-capture groups**
 
-For example:
+    `(?:thismatches) (?:[abc]([def]))|(?:[ghi])`
 
-```xml
-<transform from="\u037A\u037A" to="\u037A" error="fail" />
-```
+    Only the innermost group is allowed to be a capture group.
 
-This indicates that it is an error to type two iota subscripts immediately after each other.
+- **Disjunctions**
 
-In terms of how these different attributes work in processing a sequence of transforms, consider the transform:
+    `abc|def`
 
-```xml
-<transform before="X" from="Y" to="B" />
-```
+    Match either `abc` or `def`.
 
-This would transform the string:
+- **Match a single Unicode codepoint**
 
-```default
-XYZ → XBZ
-```
+    `.`
 
-If we mark where the current match position is before and after the transform we see:
+    Matches a codepoint, not individual code units. (See the ’u’ option in EcmaScript262 regex.)
+    For example, Osage `𐒵` is one match (`.`) not two.
+    Does not match [markers](#markers). (See `\m{.}` and `\m{marker}`, below.)
 
-```default
-X | Y Z → X B | Z
-```
+- **Match the start of the text context**
 
-And a subsequent transform could transform the Z string, looking back (using @before) to match the B.
+    `^`
 
-* * *
+    The start of the context could be the start of a line, a grid cell, or some other formatting boundary.
+    See description at the top of [`transforms`](#element-transform).
 
-### <a name="Element_reorders" href="#Element_reorders">Element: reorders</a>
+#### Additional Features
 
-This element defines a group of one or more `reorder` elements associated with this keyboard layout.
+The following are additions to standard Regex syntax.
 
-> <small>
->
-> Parents: [keyboard](#Element_keyboard)
+- **Match a Marker**
+
+    `\m{Some_Marker}`
+
+    Matches the named marker.
+    Also see [Markers](#markers).
+
+- **Match a single marker**
+
+    `\m{.}`
+
+    Matches any single marker.
+    Also see [Markers](#markers).
+
+- **Variables**
+
+    `${zwnj}`
+
+    In this usage, the variable with `id="Matras"` will be substituted in at this point in the expression. The variable can contain a range, a character, or any other portion of a pattern. If `zwnj` is a simple string, the pattern will match that string at this point.
```

Comment:
Uh... variable with `id="zwnj"`?  I don't see where the "Matras" part fits in the example here.

Reply:
thanks, good catch